ow_lander/src/ow_lander/actions.py

```diff
@@ -533,10 +583,18 @@ def execute_action(self, goal):
       trajectory = self.plan_end_effector_to_pose(intended_pose_stamped)
       self._arm.execute_arm_trajectory(trajectory,
         action_feedback_cb=self.publish_feedback_cb)
-    except (ArmExecutionError, ArmPlanningError) as err:
+    except ArmExecutionError as err:
+      self._arm.checkin_arm(self.name)
```

Comment:
Since `_arm_faults_internal_messages_publish` is not called here, you could produce an error by following a modification of the PR test:
1. `rosrun ow_lander arm_move_cartesian_guarded.py`
2. `arm_move_cartesian_guarded.py -x 1.86 -y -0.63 -z 0.045 -q 1 0.005 0.053 0.029`
3. While ArmMoveCartesian executes, in a 5th terminal call `rosrun ow_lander arm_stop.py`
Now `/system_faults_status` happens to still be true, an **ARM_GOAL_ERROR** has occurred as a result of a stop command, but the `/arm_faults_status` message still shows **TRAJECTORY_GENERATION**, which is an incorrect reason for the fault.

What I am pointing out here is that the loop of the internal fault message has to be closed somewhere in every arm action call. And by closed, I mean at some point in the course of the arm action execution the message has to be updated with either a fail or success result. Or for the purposes of this PR either **NONE** or **TRAJECTORY_GENERATION**. The loop can either be closed at the end of the next called action, as it happens here (except in the case where an execution error has occurred) or it could happen at the beginning of the next called action. It turns out `/system_faults_status` closes this loop at the end of the next called action, but I think that is wrong. 

I feel it makes more sense to close this loop at the beginning of the next called arm action. That way the `/arm_faults_status` will show that no faults are active during arm movement.